sos/report/plugins/ceph.py

```diff
@@ -138,4 +148,32 @@ def setup(self):
             "/etc/ceph/*bindpass*"
         ])
 
+        # If containerized, run commands in containers
+        containers_list = self.list_ceph_containers("ceph-*")
+        if containers_list:
+            # Avoid retrieving multiple times the same data
+            got_ceph_cmds = False
+            for container in containers_list:
+                if re.match("ceph-(mon|rgw|osd)", container[1]) and \
+                        not got_ceph_cmds:
+                    self.add_cmd_output([
+                        self.fmt_container_cmd(container[1], "ceph %s" % s)
+                        for s in ceph_cmds
+                    ])
+                    got_ceph_cmds = True
+                if re.match("ceph-osd", container[1]):
+                    self.add_cmd_output([
+                        self.fmt_container_cmd(container[1], "%s" % s)
+                        for s in ceph_osd_cmds
+                    ])
+                    break
```

Comment:
Why break here? If we iterate over `ceph-osd` before `ceph-mon` then we'd skip collection of the latter.

Reply:
Because if the first container in the list is an OSD then both list of commands will be retrieved and the iteration will stop (which is the expected result). If it is another kind of container (mon or mgr) it will retrieve "ceph_cmds" and continue iterate until it finds (or not) an OSD container and gets commands specific to OSDs (ceph_osd_cmds) then stop. I did put the break there it is because there can be a lot of OSD containers running (minimum 1 / disk on the server) and running "ceph_osd_cmds" on all of them will be useless as the output will always be the same.

Reply:
Ack.